clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java

```diff
@@ -172,11 +174,14 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
             Put chunkPut = new Put();
             chunkPut.putValue = chunkValue;
             chunkPut.schemaId = schemaId;
+            if (hasReplicationMetadata) {
+              chunkPut.replicationMetadataPayload = ByteBuffer.allocate(0);
+              chunkPut.replicationMetadataVersionId = -1;
+            }
```

Comment:
Could you please share details on why we need this branch? Will we replace this code piece later?

Reply:
Yes, this is basically the same logic as in the VeniceWriter#putLargeRecord.
In the VeniceWriter:
1. When the store is not A/A, then there is no metadata passed in, so don't set it.
2. When the store is in A/A, then there is metadata, but the chunk value does not carry metadata so just set as default value. Maybe I should use constant to make it more clear.

This is just to be consistent with VW logic, as VW produces the record to the VT for follower to consume. We should let leader and follow has exactly the same data.

Reply:
Please take a look with my latest update. I added/referenced constant for this.
Maybe potentially we could refactor this part in the near future to let it and VW reuse same util method.